### DISCUSSION TITLE
is the unlocker GPL?

### DISCUSSION THREAD

**Discussion by Cuteistfox** [Hamza417/Inure#163] [Q&A] [answered]
The app is so the unlocker ***should*** be but is it

**Comment by Hamza417** [comment]
No, the unlocker is a _closed source_ application.

**Reply by Cuteistfox** [reply]
but thhe app is GPL

**Reply by Hamza417** [reply]
Yes, but the unlocker does not have to be. It's a separate app.

**Reply by Hamza417** [reply]
Do you need the Unlocker to be open sourced?

**Reply by Cuteistfox** [reply]
i have  strict no proprietary software police on my fire(expet amazon bloat(i disable butt i cant uninstall till someone manages to unlock bootloader and set up TWRP/Fastboot))

**Reply by Hamza417** [reply, marked as answer]
I made the Unlocker public under [Apache 2.0 license](https://github.com/Hamza417/InureUnlocker/blob/master/LICENSE), you can check the source code now. Does that help you?

**Reply by Cuteistfox** [reply]
i would recomend you make it pivate and only distribute it when inure unlocker is purchased

**Reply by Hamza417** [reply]
The license doesn't approve that, so it'd be best to leave it public.

**Reply by Cuteistfox** [reply]
yes it does allow  see https://www.gnu.org/philosophy/selling.html

**Reply by Hamza417** [reply]
> The one exception is in the case where binaries are distributed without the corresponding complete source code. Those who do this are required by the GNU GPL to provide source code on subsequent request. 

Yes, this approves, thanks for the update. I'll update the repo's visibility and let me know when you require the source code.

**Reply by Cuteistfox** [reply]
you have to add a notice to inure unlocker and the purchase page on how to request source

**Reply by Cuteistfox** [reply]
> > The one exception is in the case where binaries are distributed without the corresponding complete source code. Those who do this are required by the GNU GPL to provide source code on subsequent request.
> 
> Yes, this approves, thanks for the update. I'll update the repo's visibility and let me know when you require the source code.

thats only for (A/L)GPL

**Reply by Hamza417** [reply]
> you have to add a notice to inure unlocker and the purchase page on how to request source.

Yes, I'm aware but it'll take sometime

> thats only for (A/L)GPL

I'll have to check for Apache first, otherwise I'll change the license to GPLv3

**Reply by mdaniel** [reply]
> Yes, this approves, thanks for the update. I'll update the repo's visibility and let me know when you require the source code.

I have purchased the unlocker, and don't necessarily need the source code but wanted to open an issue in the repo to suggest that the unlocker itself not have a launcher icon. It's distracting having the two icons right next to each other, when there's no UI for the unlocker and clicking on it doesn't do anything helpful. As an example of what I'm suggesting, "Sleep as Android" has both an Unlock and SleepCloud add-on, and when either of them are purchased they show up in the Inure list because they're "installed" but they don't show up in the launcher

I'm also open to putting the request in the Ideas discussion group for Inure itself, if that'd be more helpful

**Reply by Hamza417** [reply]
There's an option to hide the icon in the unlocker but for some reason it's broken in Android 13 and above, you could try that and see if it works?

I'm also planning to move to IAP model and eventually get rid of the unlocker in the next few months.

**Reply by mdaniel** [reply]
> There's an option to hide the icon in the unlocker

As I said, when I click on the unlocker icon it just launches to the Android "App info" screen, which doesn't show any hide options. But if you're going to transition away from the unlocker, then I'll just wait for that

**Reply by Cuteistfox** [reply]
How would IAP work without play services across multiple devices?

**Reply by Hamza417** [reply]
> How would IAP work without play services across multiple devices?

It'd be exclusive to play builds, github builds already support both activation with licence key and the unlocker.

**Reply by Cuteistfox** [reply]
licence key would be easy to extract or it would not work on multiple devices.
